nix/buildLeanPackage.nix

```diff
@@ -154,17 +164,19 @@ in rec {
       ${lib.concatStringsSep " " (map (d: "${d}/*.a") allStaticLibDeps)} \
```

Comment:
You're not actually linking against `nativeSharedLibs` here? Doing so correctly might actually be tricky - if we built a dependency into a shared library, we must not link against it again in `allStaticLibDeps`...

Reply:
Yes, that might be tricky, but that might be solved with some extra option to exclude certain static libraries from extra linking.